### ISSUE TITLE
Accessibility: Interactive elements nested in an invalid way

### ISSUE THREAD

**Issue by ChloeSartorelli** [silverstripe/silverstripe-asset-admin#1593]
[WCAG 2.2 A 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
[ARIA 1.2 Presentational Children 5.2.9](https://www.w3.org/TR/wai-aria-1.2/#childrenArePresentational)
Giving an element an interactive `role` attribute (e.g. `<div role="button"></div>` marks child elements as presentational, which hides them from the accessibility tree.

The files and folders in the gallery view have a div that looks like:
`<div data-draggable='true' role='button' tabindex='0' aria-disabled='false' aria-roledescription='draggable' aria-describedby='DndDescribedBy-0'>...</div>`

These contain an `<input>` which is the checkbox for selecting multiple items. This means that the draggable `div` will be announced, but or some screen readers even though users _can tab to_ the selection checkbox, the screen reader won't announce it.

It also makes it difficult to select and unselect items using the keyboard.

## Possible solutions

- Use a specific drag handle for keyboard-interactive drag/drop functionality instead of having the whole outer element selectable
- Remove keyboard interaction directly on the file/folder for drag/drop functionality - users can use the menu bar at the bottom of the screen after selecting a file or folder.

## Acceptance criteria

- [ ] The interactive draggable element doesn't contain secondary interactive elements (or is itself not interactive with the keyboard and that's made clear to assistive technologies)
- [ ] The following actions can be performed using both mouse and keyboard on files and folders:
    - [ ] select (e.g. if you want to use batch actions)
    - [ ] move (e.g. move file into new folder)
    - [ ] edit
- [ ] It's clear when you are focusing on the select checkbox, and when you are focusing on the drag element (or the drag element is not focusable)
- [ ] The changelog clearly calls out any changes to markup selectors so people can adjust their custom css/js as necessary
